zubhub_frontend/zubhub/src/components/upload_file/uploadFile.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import clsx from 'clsx';
+import ImageIcon from '@material-ui/icons/Image';
+import CustomButton from '../button/Button';
+import {
+  handleFileButtonClick,
+  handleImageFieldChange,
+} from './uploadFileScripts.js';
+import { Typography, FormControl, FormHelperText } from '@material-ui/core';
+import { getFieldAndIndex } from '../../assets/js/utils/scripts';
+
+function UploadFile(props) {
+  const {
+    id,
+    fileType,
+    uploadButtonLabel,
+    classes,
+    countFilesText,
+    multiple,
+    wraperState,
+  } = props;
+
+  const { field, index } = getFieldAndIndex(id);
```

Comment:
there should be a better way to get the field info instead of splitting the id.

Reply:
would you please give some hints, here i was just trying to mimic Formik behavior for fields named with indexes (field[index])